dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -195,17 +193,19 @@ public void close() {
         private static final Logger log = LoggerFactory.getLogger(EncodingDeserializerRegistry.class);
         private final ImmutableList<EncodingDeserializerContainer<T>> encodings;
         private final ErrorDecoder errorDecoder;
-        private final TypeMarker<T> token;
-        private final boolean isOptionalType;
         private final Optional<String> acceptValue;
+        private final Supplier<T> emptyInstance;
 
-        EncodingDeserializerRegistry(List<Encoding> encodings, ErrorDecoder errorDecoder, TypeMarker<T> token) {
+        EncodingDeserializerRegistry(
+                List<Encoding> encodings,
+                ErrorDecoder errorDecoder,
+                EmptyContainerDeserializer empty,
+                TypeMarker<T> token) {
             this.encodings = encodings.stream()
                     .map(encoding -> new EncodingDeserializerContainer<>(encoding, token))
                     .collect(ImmutableList.toImmutableList());
             this.errorDecoder = errorDecoder;
-            this.token = token;
-            this.isOptionalType = TypeMarkers.isOptional(token);
+            this.emptyInstance = Suppliers.memoize(() -> empty.getEmptyInstance(token)); // TODO(dfox): save exception?
```

Comment:
I think by memoizing we will lose the exception if mulitple calls receive a 204 for a non empty type. can we construct the instance at response time?

Reply:
I'd prefer not to construct the instance every time because we always expect to get the same answer

Reply:
I was wrong, memoizing won't interfere with the exception so this works well